packages/rocketchat-ldap/server/sync.js

```diff
@@ -84,14 +85,19 @@ getDataToSyncUserData = function getDataToSyncUserData(ldapUser, user) {
 					}
 					break;
 
-				case 'name':
+				default:
+					if (!_.find(whitelistedUserFields, (el) => el === userField.split('.')[0])) {
+						logger.debug(`user attribute not whitelisted: ${ userField }`);
+						return;
+					}
+
 					const tmpLdapField = RocketChat.templateVarHandler(ldapField, ldapUser.object);
+					const userFieldValue = _.reduce(userField.split('.'), (acc, el) => acc[el], user);
 
-					if (tmpLdapField && user.name !== tmpLdapField) {
-						userData.name = tmpLdapField;
-						logger.debug(`user.name changed to: ${ tmpLdapField }`);
+					if (tmpLdapField && userFieldValue !== tmpLdapField) {
+						userData[userField] = tmpLdapField;
```

Comment:
@rodrigok 

In my tests, if the variable userData contains the object `{ name: 'value1', 'customFields.secao-descricao': 'value2','customFields.secao-sigla': 'value3' }`, the object stored in the users collection, inside a user document is `"customFields" : { "secao-sigla" : "value3", "secao-descricao" : "value2" }`. No exceptions occur.

We have other examples in the same code (sync.js), in the line 123, for instance: `userData['services.ldap.id'] = uniqueId.value;`.

Maybe I still didn't understand what you're trying to show me. :-(